src/components/Home.jsx

```diff
@@ -1,7 +1,44 @@
 import React from 'react';
+import getToken from '../lib/tokens';
+import { Redirect } from 'react-router-dom';
+import { useHistory } from 'react-router-dom';
 
 const Home = () => {
-  return <h1>Home</h1>;
+  // const [hasToken, setHasToken] = useState(false);
+
+  let history = useHistory();
+
+  const Token = function generateToken() {
```

Comment:
Consider using an arrow function here to be able to use the `generateToken` name in the component.

```jsx
const generateToken = () => {
```
Then in the component, it can be called by:
```jsx
onClick={generateToken}
```

Reply:
Adds more clarity to the code, so I've implemented your suggestion. Thank you